File: src/main/kotlin/org/voegtle/wunschmanager/data/WishList.kt
package org.voegtle.wunschmanager.data

import com.googlecode.objectify.annotation.Entity
import com.googlecode.objectify.annotation.Id
import com.googlecode.objectify.annotation.Index

@Entity
class WishList(@Id var id: Long? = null, @Index var owner: String? = null, @Index var event: String = "",
               @Index var description: String = "", var managed: Boolean = false,
               @Index var createTimestamp: Long? = null) : Comparable<WishList> {
  override fun compareTo(other: WishList): Int {
    return when {
      createTimestamp ?: 0 < other.createTimestamp ?: 0 -> -1
      createTimestamp ?: 0 > other.createTimestamp ?: 0 -> 1
      else -> 0
    }
  }

}

File: frontend/src/app/services/wish-list.ts

export interface WishList {
  id: number;
  owner: string;
  event: string;
  description: string;
  managed: boolean;
  background: string;
}

export class WishListImpl implements WishList {
  background: string;
  description: string;
  event: string;
  id: number;
  managed: boolean;
  owner: string;

  public copy(source: WishList) {
    this.background = source.background;
    this.description = source.description;
    this.event = source.event;
    this.managed = source.managed;
  }
}
